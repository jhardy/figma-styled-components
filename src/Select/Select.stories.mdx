import { Select } from './Select'
import { Story, Meta } from '@storybook/addon-docs/blocks'

<Meta title='MDX|Select' component={Select} />

# Select

## Default Select

<Story name='Select'>
  <div className='component-box'>
    <Select
      options={[
        { value: '1', label: 'One' },
        { value: '2', label: 'Two' },
        { value: '3', label: 'Three' }
      ]}
    />
  </div>
</Story>

## Select with Groups

<Story name='Select with groups'>
  <div className='component-box'>
    <Select
      placeholder='Select an option'
      options={[
        {
          label: 'Group 1',
          group: [
            { label: 'Item 1', value: '1' },
            { label: 'Item 2', value: '2' }
          ]
        },
        {
          label: 'Group 2',
          group: [
            { label: 'Item 3', value: '3' },
            { label: 'Item 4', value: '4' }
          ]
        },
        {
          label: 'Group 3',
          group: [
            { label: 'Item 5', value: '5' },
            { label: 'Item 6', value: '6' }
          ]
        }
      ]}
    />
  </div>
</Story>

<div className="props-table">

## Props

| Prop Name   | Type                                                                                           | Default | Required | Description                             |
| :---------- | :--------------------------------------------------------------------------------------------- | :-----: | :------: | :-------------------------------------- |
| options     | <code>Array<{label: string, value?: string, group?:Array<value:string, label:string> }></code> |    -    |   yes    | Array of options for the select         |
| placeholder | string                                                                                         |    -    |    -     | Placeholder for the select              |
| onChange    | function, returns a string of the selected value                                               |    -    |    -     | Callback for when an select has changed |

</div>
