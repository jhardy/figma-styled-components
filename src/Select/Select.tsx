import * as React from 'react'
import styled from 'styled-components'

import { Text } from '../Text'

export interface SelectOptionItem {
  label: string
  value?: string
}

export interface SelectOptions extends SelectOptionItem {
  group?: SelectOptionItem[]
}

export interface SelectProps {
  options: SelectOptions[]
  onChange?: any
  placeholder?: string
  noDefault?: boolean
}

const SelectedCheck = () => {
  return (
    <svg
      width='16'
      height='16'
      viewBox='0 0 16 16'
      fill='none'
      xmlns='http://www.w3.org/2000/svg'
    >
      <path
        fillRule='evenodd'
        clipRule='evenodd'
        d='M13.2069 5.20724L7.70688 10.7072L6.99977 11.4144L6.29267 10.7072L3.29267 7.70724L4.70688 6.29303L6.99977 8.58592L11.7927 3.79303L13.2069 5.20724Z'
        fill='white'
        fillOpacity='1'
      />
    </svg>
  )
}

const SelectItem = styled.li<{ selected?: boolean }>`
  box-sizing: border-box;
  display: flex;
  align-items: center;
  width: 100%;
  height: 24px;
  padding: 0 16px 0 32px;
  list-style-type: none;
  text-align: left;
  cursor: pointer;
  position: relative;

  svg {
    position: absolute;
    left: 6px;
    top: 4px;
    pointer-events: none;
  }

  &:hover {
    background-color: #18a0fb;
  }
`

const SelectGroup = styled.div`
`

const SelectChevronIcon = (props: any) => {
  return(
  <div {...props}><svg width='8' height='5' viewBox='0 0 8 5' fill='none' xmlns='http://www.w3.org/2000/svg'>
  <path fillRule='evenodd' clipRule='evenodd' d='M3.64645 4.35359L0.646454 1.35359L1.35356 0.646484L4.00001 3.29293L6.64645 0.646484L7.35356 1.35359L4.35356 4.35359L4.00001 4.70714L3.64645 4.35359Z' fill='currentColor' />
  </svg>
  </div>
  )
}

const SelectChevron = styled(SelectChevronIcon)`
`

const SelectTrigger = styled.button`
  display: flex;
  align-items: center;
  width: 100%;
  height: 30px;
  margin: 1px 0 1px 0 !important;
  padding: 0 8px 0 8px;
  text-align: left;
  cursor: pointer;
  color: rgba(0, 0, 0, 0.8);
  border: 1px solid transparent;
  border-radius: 2px;
  background-color: #ffffff;
  font-family: "Inter", sans-serif;
  font-weight: 400;
  font-size: 11px;
  line-height: 16px;
  letter-spacing: 0.005em;

  ${SelectChevron} {
    margin-left: 8px;
    color: rgba(0, 0, 0, 0.2);
  }

  &:hover {
    border: 1px solid rgba(0, 0, 0, 0.1);
    justify-content: space-between;
    ${SelectChevron} {
      color: rgba(0, 0, 0, 1);
    }
  }

  &:focus {
    padding: 0 7px 0 7px;
    border: 2px solid #18a0fb;
    outline: none;
    justify-content: space-between;

    ${SelectChevron} {
      color: rgba(0, 0, 0, 1);
    }
  }
`

export const SelectFactory: React.FC<SelectProps> = ({
  options,
  onChange,
  placeholder,
  noDefault,
  ...props
}) => {
  const selectInput = React.useRef<HTMLSelectElement>()

  const getInitalValue = (optionItems: SelectOptions[]) => {
    if (optionItems[0].group) {
      return optionItems[0].group[0].label ? { value: optionItems[0].group[0].value, label: optionItems[0].group[0].label } : { value: optionItems[0].group[0].value, label: optionItems[0].group[0].value }
    } else {
      return optionItems[0].label ? { value: optionItems[0].value, label: optionItems[0].label } : { value: optionItems[0].value, label: optionItems[0].value }
    }

  }

  const [selectedOption, setOption] = React.useState(getInitalValue(options))
  const [hasMadeSelection, setHasMadeSelection] = React.useState(placeholder ? false : true)
  const [showOptions, setOptionsVisiblity] = React.useState(false)

  const handleClick = (event: React.MouseEvent) => {
    setHasMadeSelection(true)
    // const ensureLabel = label === '' ? value : label
    const target = event.currentTarget
    // console.log(target, target.getAttribute('data-value'))
    const value = target.getAttribute('data-value') || ''
    const label = target.getAttribute('data-label') || ''

    setOption({ value, label })
    setOptionsVisiblity(false)
  }


  const wrapperRef = React.useRef(null)

  const toggleSelect = (e: any) => {
    setOptionsVisiblity(!showOptions)

  }

  function useOutsideAlerter (ref: any) {

    function handleClickOutside (event: any) {
      if (ref.current && !ref.current.contains(event.target)) {

        toggleSelect(event)
      }
    }

    React.useEffect(() => {
      // Bind the event listener
      document.addEventListener('mousedown', handleClickOutside)
      return () => {
        // Unbind the event listener on clean up
        document.removeEventListener('mousedown', handleClickOutside)
      }
    })
  }

  useOutsideAlerter(wrapperRef)


  return (
    <div {...props} ref={wrapperRef}>

      <select
        ref={selectInput as any/* this is :(, fix soon!*/}
        onChange={onChange}
        defaultValue={placeholder ? '' : selectedOption.value}
      >
        {noDefault && <option />}
        {options.map((option) => {
          return option.group ?
            <optgroup label={option.label} key={`opt-group-` + option.label}>
              {option.group.map((item) => {
                return <option key={`group` + item.label} value={item.value || item.label}>{item.label}</option>
              })}
            </optgroup>
            :
            <option key={option.label} value={option.value || option.label}>
            {option.label}
            </option>
        })}
      </select>
      <SelectTrigger onClick={toggleSelect}>
        <Text>{placeholder && !hasMadeSelection ? placeholder : selectedOption.label}</Text>
        <SelectChevron />
      </SelectTrigger>
      <ul className={showOptions ? 'show-options' : undefined}>
        {options.map((option, i) => {
          return option.group ?
            <SelectGroup key={`group-parent-` + i}>
              {option.group.map((item) => {
                return (
                <SelectItem key={`group-` + item.label} id={item.value || item.label} data-value={item.value} data-label={item.label || item.value} onClick={handleClick}>
                  {selectedOption.value === item.value && hasMadeSelection && <SelectedCheck />}
                  <Text size='ui12' inverted={true}>
                    {item.label}
                  </Text>
                </SelectItem>)
              })}
            </SelectGroup>
            :
            <SelectItem
              key={`list-` + i}
              id={option.value}
              data-value={option.value}
              data-label={option.label || option.value}
              onClick={handleClick}
            >
              {selectedOption.value === option.value && <SelectedCheck />}
              <Text size='ui12' inverted={true}>
                {option.label || option.value}
              </Text>
            </SelectItem>

        })}
      </ul>
    </div>
  )
}

export const Select = styled(SelectFactory)`
  position: relative;
  display: flex;
  flex-direction: row;
  flex-grow: 2;
  flex-wrap: nowrap;
  width: 100%;

  select {
    display: none;
  }

  ul {
    position: absolute;
    z-index: 2;
    top: 31px;
    right: 0;
    left: 0;
    display: block;
    overflow: auto;
    width: 100%;
    margin: 0;
    padding: 8px 0 8px 0;
    list-style-type: none;
    opacity: 0;
    box-shadow: 0 5px 17px rgba(0, 0, 0, 0.2), 0 2px 7px rgba(0, 0, 0, 0.15);
    background-color: #222222;
    display: none;

    &.show-options {
      opacity: 1;
      display: block;
    }
  }

  ${SelectGroup}:not(:first-child) {
    padding: 8px 0 0 0;
    margin-top: 8px;
    border-top: solid 1px rgba(255, 255, 255, 0.2);
  }

  ${SelectGroup}:last-child {
    padding-bottom: 0;
  }
`
