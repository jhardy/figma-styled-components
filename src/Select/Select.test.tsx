import 'jest-styled-components'
import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { Select } from './Select'


const options = [{ label: 'Option One', value: 'one' },
{ label: 'Option Two', value: 'two' },
{ label: 'Option Three', value: 'three' }]

const groupedOptions = [{
  label: 'Group 1',
  // tslint:disable-next-line: object-literal-sort-keys
  group: [
    { label: 'Item 1', value: '1' },
    { label: 'Item 2', value: '2' }
  ]
},
  {
    label: 'Group 2',
    // tslint:disable-next-line: object-literal-sort-keys
    group: [
    { label: 'Item 3', value: '3' },
    { label: 'Item 4', value: '4' }
    ]
  },
  {
    label: 'Group 3',
    // tslint:disable-next-line: object-literal-sort-keys
    group: [
    { label: 'Item 5', value: '5' },
    { label: 'Item 6', value: '6' }
    ]
  }
]
test('A default Select', () => {
  const tree = renderer.create(<Select options={options} />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('A Select with grouped options', () => {
  const tree = renderer.create(<Select options={groupedOptions} />).toJSON()
  expect(tree).toMatchSnapshot()
})



