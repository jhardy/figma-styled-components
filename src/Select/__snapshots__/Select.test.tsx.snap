// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A Select with grouped options 1`] = `
.c2 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: "Inter",sans-serif;
  color: rgba(0,0,0,0.8);
  font-weight: 400;
  font-size: 11px;
  -webkit-letter-spacing: 0.005em;
  -moz-letter-spacing: 0.005em;
  -ms-letter-spacing: 0.005em;
  letter-spacing: 0.005em;
  line-height: 16px;
}

.c6 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: "Inter",sans-serif;
  color: #ffffff;
  font-weight: 400;
  font-size: 12px;
  -webkit-letter-spacing: .005em;
  -moz-letter-spacing: .005em;
  -ms-letter-spacing: .005em;
  letter-spacing: .005em;
  line-height: 16px;
}

.c5 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 24px;
  padding: 0 16px 0 32px;
  list-style-type: none;
  text-align: left;
  cursor: pointer;
  position: relative;
}

.c5 svg {
  position: absolute;
  left: 6px;
  top: 4px;
  pointer-events: none;
}

.c5:hover {
  background-color: #18a0fb;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 30px;
  margin: 1px 0 1px 0 !important;
  padding: 0 8px 0 8px;
  text-align: left;
  cursor: pointer;
  color: rgba(0,0,0,0.8);
  border: 1px solid transparent;
  border-radius: 2px;
  background-color: #ffffff;
  font-family: "Inter",sans-serif;
  font-weight: 400;
  font-size: 11px;
  line-height: 16px;
  -webkit-letter-spacing: 0.005em;
  -moz-letter-spacing: 0.005em;
  -ms-letter-spacing: 0.005em;
  letter-spacing: 0.005em;
}

.c1 .c3 {
  margin-left: 8px;
  color: rgba(0,0,0,0.2);
}

.c1:hover {
  border: 1px solid rgba(0,0,0,0.1);
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1:hover .c3 {
  color: rgba(0,0,0,1);
}

.c1:focus {
  padding: 0 7px 0 7px;
  border: 2px solid #18a0fb;
  outline: none;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1:focus .c3 {
  color: rgba(0,0,0,1);
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex: 2;
  -webkit-flex-grow: 2;
  -ms-flex-positive: 2;
  flex-grow: 2;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  width: 100%;
}

.c0 select {
  display: none;
}

.c0 ul {
  position: absolute;
  z-index: 2;
  top: 31px;
  right: 0;
  left: 0;
  display: block;
  overflow: auto;
  width: 100%;
  margin: 0;
  padding: 8px 0 8px 0;
  list-style-type: none;
  opacity: 0;
  box-shadow: 0 5px 17px rgba(0,0,0,0.2),0 2px 7px rgba(0,0,0,0.15);
  background-color: #222222;
  display: none;
}

.c0 ul.show-options {
  opacity: 1;
  display: block;
}

.c0 .c4:not(:first-child) {
  padding: 8px 0 0 0;
  margin-top: 8px;
  border-top: solid 1px rgba(255,255,255,0.2);
}

.c0 .c4:last-child {
  padding-bottom: 0;
}

<div
  className="c0"
>
  <select
    defaultValue="1"
  >
    <optgroup
      label="Group 1"
    >
      <option
        value="1"
      >
        Item 1
      </option>
      <option
        value="2"
      >
        Item 2
      </option>
    </optgroup>
    <optgroup
      label="Group 2"
    >
      <option
        value="3"
      >
        Item 3
      </option>
      <option
        value="4"
      >
        Item 4
      </option>
    </optgroup>
    <optgroup
      label="Group 3"
    >
      <option
        value="5"
      >
        Item 5
      </option>
      <option
        value="6"
      >
        Item 6
      </option>
    </optgroup>
  </select>
  <button
    className="c1"
    onClick={[Function]}
  >
    <span
      className="c2"
    >
      Item 1
    </span>
    <div
      className="c3 "
    >
      <svg
        fill="none"
        height="5"
        viewBox="0 0 8 5"
        width="8"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M3.64645 4.35359L0.646454 1.35359L1.35356 0.646484L4.00001 3.29293L6.64645 0.646484L7.35356 1.35359L4.35356 4.35359L4.00001 4.70714L3.64645 4.35359Z"
          fill="currentColor"
          fillRule="evenodd"
        />
      </svg>
    </div>
  </button>
  <ul>
    <div
      className="c4 "
    >
      <li
        className="c5"
        id="1"
        onClick={[Function]}
      >
        <svg
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M13.2069 5.20724L7.70688 10.7072L6.99977 11.4144L6.29267 10.7072L3.29267 7.70724L4.70688 6.29303L6.99977 8.58592L11.7927 3.79303L13.2069 5.20724Z"
            fill="white"
            fillOpacity="1"
            fillRule="evenodd"
          />
        </svg>
        <span
          className="c6"
          onNegative={true}
          size="medium"
        >
          Item 1
        </span>
      </li>
      <li
        className="c5"
        id="2"
        onClick={[Function]}
      >
        <span
          className="c6"
          onNegative={true}
          size="medium"
        >
          Item 2
        </span>
      </li>
    </div>
    <div
      className="c4 "
    >
      <li
        className="c5"
        id="3"
        onClick={[Function]}
      >
        <span
          className="c6"
          onNegative={true}
          size="medium"
        >
          Item 3
        </span>
      </li>
      <li
        className="c5"
        id="4"
        onClick={[Function]}
      >
        <span
          className="c6"
          onNegative={true}
          size="medium"
        >
          Item 4
        </span>
      </li>
    </div>
    <div
      className="c4 "
    >
      <li
        className="c5"
        id="5"
        onClick={[Function]}
      >
        <span
          className="c6"
          onNegative={true}
          size="medium"
        >
          Item 5
        </span>
      </li>
      <li
        className="c5"
        id="6"
        onClick={[Function]}
      >
        <span
          className="c6"
          onNegative={true}
          size="medium"
        >
          Item 6
        </span>
      </li>
    </div>
  </ul>
</div>
`;

exports[`A default Select 1`] = `
.c2 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: "Inter",sans-serif;
  color: rgba(0,0,0,0.8);
  font-weight: 400;
  font-size: 11px;
  -webkit-letter-spacing: 0.005em;
  -moz-letter-spacing: 0.005em;
  -ms-letter-spacing: 0.005em;
  letter-spacing: 0.005em;
  line-height: 16px;
}

.c5 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: "Inter",sans-serif;
  color: #ffffff;
  font-weight: 400;
  font-size: 12px;
  -webkit-letter-spacing: .005em;
  -moz-letter-spacing: .005em;
  -ms-letter-spacing: .005em;
  letter-spacing: .005em;
  line-height: 16px;
}

.c4 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 24px;
  padding: 0 16px 0 32px;
  list-style-type: none;
  text-align: left;
  cursor: pointer;
  position: relative;
}

.c4 svg {
  position: absolute;
  left: 6px;
  top: 4px;
  pointer-events: none;
}

.c4:hover {
  background-color: #18a0fb;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 30px;
  margin: 1px 0 1px 0 !important;
  padding: 0 8px 0 8px;
  text-align: left;
  cursor: pointer;
  color: rgba(0,0,0,0.8);
  border: 1px solid transparent;
  border-radius: 2px;
  background-color: #ffffff;
  font-family: "Inter",sans-serif;
  font-weight: 400;
  font-size: 11px;
  line-height: 16px;
  -webkit-letter-spacing: 0.005em;
  -moz-letter-spacing: 0.005em;
  -ms-letter-spacing: 0.005em;
  letter-spacing: 0.005em;
}

.c1 .c3 {
  margin-left: 8px;
  color: rgba(0,0,0,0.2);
}

.c1:hover {
  border: 1px solid rgba(0,0,0,0.1);
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1:hover .c3 {
  color: rgba(0,0,0,1);
}

.c1:focus {
  padding: 0 7px 0 7px;
  border: 2px solid #18a0fb;
  outline: none;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1:focus .c3 {
  color: rgba(0,0,0,1);
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex: 2;
  -webkit-flex-grow: 2;
  -ms-flex-positive: 2;
  flex-grow: 2;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  width: 100%;
}

.c0 select {
  display: none;
}

.c0 ul {
  position: absolute;
  z-index: 2;
  top: 31px;
  right: 0;
  left: 0;
  display: block;
  overflow: auto;
  width: 100%;
  margin: 0;
  padding: 8px 0 8px 0;
  list-style-type: none;
  opacity: 0;
  box-shadow: 0 5px 17px rgba(0,0,0,0.2),0 2px 7px rgba(0,0,0,0.15);
  background-color: #222222;
  display: none;
}

.c0 ul.show-options {
  opacity: 1;
  display: block;
}

.c0 .c6:not(:first-child) {
  padding: 8px 0 0 0;
  margin-top: 8px;
  border-top: solid 1px rgba(255,255,255,0.2);
}

.c0 .c6:last-child {
  padding-bottom: 0;
}

<div
  className="c0"
>
  <select
    defaultValue="one"
  >
    <option
      value="one"
    >
      Option One
    </option>
    <option
      value="two"
    >
      Option Two
    </option>
    <option
      value="three"
    >
      Option Three
    </option>
  </select>
  <button
    className="c1"
    onClick={[Function]}
  >
    <span
      className="c2"
    >
      Option One
    </span>
    <div
      className="c3 "
    >
      <svg
        fill="none"
        height="5"
        viewBox="0 0 8 5"
        width="8"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M3.64645 4.35359L0.646454 1.35359L1.35356 0.646484L4.00001 3.29293L6.64645 0.646484L7.35356 1.35359L4.35356 4.35359L4.00001 4.70714L3.64645 4.35359Z"
          fill="currentColor"
          fillRule="evenodd"
        />
      </svg>
    </div>
  </button>
  <ul>
    <li
      className="c4"
      id="one"
      onClick={[Function]}
    >
      <svg
        fill="none"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M13.2069 5.20724L7.70688 10.7072L6.99977 11.4144L6.29267 10.7072L3.29267 7.70724L4.70688 6.29303L6.99977 8.58592L11.7927 3.79303L13.2069 5.20724Z"
          fill="white"
          fillOpacity="1"
          fillRule="evenodd"
        />
      </svg>
      <span
        className="c5"
        onNegative={true}
        size="medium"
      >
        Option One
      </span>
    </li>
    <li
      className="c4"
      id="two"
      onClick={[Function]}
    >
      <span
        className="c5"
        onNegative={true}
        size="medium"
      >
        Option Two
      </span>
    </li>
    <li
      className="c4"
      id="three"
      onClick={[Function]}
    >
      <span
        className="c5"
        onNegative={true}
        size="medium"
      >
        Option Three
      </span>
    </li>
  </ul>
</div>
`;
